compiler-core/src/type_/environment.rs

```diff
@@ -393,6 +393,7 @@ impl<'a> Environment<'a> {
         &mut self,
         module: Option<&EcoString>,
         name: &EcoString,
+        situation: Option<UnknownValueConstructorErrorSituation>,
```

Comment:
Rather than pass in a situation let's have it default to non and then the calling code can optionally add it using `map_err`, the same as it does for the unify errors.